### ISSUE TITLE
Can not install when xcode's app name contains space.

### ISSUE THREAD

**Issue by rogierdna** [realm/jazzy#994]
Hello, there's an issue with jazzy installation when current xcode's application name contains space. ie "Xcode-beta 3.app"

(Renaming xcode app name so that it contains no spaces, fixes the issue as a workaround.)

With space in app name, jazzy Installation fails with  
/bin/sh: /Applications/Xcode-beta: No such file or directory

sudo gem install jazzy
Password:
Building native extensions.  This could take a while...
ERROR:  Error installing jazzy:
	ERROR: Failed to build gem native extension.

    current directory: /Users/xx/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/ffi-1.9.25/ext/ffi_c
/Users/xx/.rbenv/versions/2.4.1/bin/ruby -r ./siteconf20180717-58080-1py5c37.rb extconf.rb
checking for ffi.h... no
checking for ffi.h in /usr/local/include,/usr/include/ffi... yes
checking for ffi_call() in -lffi... yes
checking for ffi_closure_alloc()... no
checking for ffi_raw_call()... no
checking for shlwapi.h... no
checking for ruby/thread.h... yes
checking for rb_thread_blocking_region()... no
checking for rb_thread_call_with_gvl()... yes
checking for rb_thread_call_without_gvl()... yes
creating extconf.h
creating Makefile

current directory: /Users/xx/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/ffi-1.9.25/ext/ffi_c
make "DESTDIR=" clean

current directory: /Users/xx/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/ffi-1.9.25/ext/ffi_c
make "DESTDIR="
Configuring libffi
clang: error: unsupported option '-print-multi-os-directory'
clang: error: no input files
cd "/Users/xx/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/ffi-1.9.25/ext/ffi_c/libffi-x86_64-darwin16" && /Applications/Xcode-beta 3.app/Contents/Developer/usr/bin/make
/bin/sh: /Applications/Xcode-beta: No such file or directory
make: *** ["/Users/xx/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/ffi-1.9.25/ext/ffi_c/libffi-x86_64-darwin16"/.libs/libffi_convenience.a] Error 127

**Comment by reececomo** [comment]
Did you try escaping the space? Does this resolve the issue

`"Xcode-beta\ 3.app"`

**Comment by johnfairh** [comment]
tldr: this is not something that can be practically fixed. I'll update the docs to document the need to avoid spaces in the Xcode path when installing jazzy.

Currently two deps have native extensions: redcarpet and ffi.  If you have a space in your Xcode then neither build on Ruby 2.3; on later Rubies only ffi fails.

Ruby's internal mkmf library has a fix post-2.3 to do with quoting pathnames that accounts for redcarpet.

Ffi though has a more complicated structure involving a recursive invocation of make expressed by the extremely normal `cd whatever && $(MAKE)`.  Unfortunately on macOS, even though stock `make` is invoked via `/usr/bin/make` the actual program is inside the Xcode developer directory and `$(MAKE)` expands to its path including any spaces in the Xcode app directory name.

**Comment by johnfairh** [comment]
Added to readme in master.
